src/oemof/solph/components/_generic_storage.py

```diff
@@ -1729,12 +1755,15 @@ def _objective_expression(self):
                         n=lifetime,
                         wacc=interest,
                     )
-                    investment_costs_increment = (
-                        self.invest[n, p]
-                        * annuity
-                        * lifetime
-                        * ((1 + m.discount_rate) ** (-m.es.periods_years[p]))
+                    duration = min(
+                        end_of_optimization - m.es.periods_years[p], lifetime
+                    )
+                    present_value_factor = 1 / economics.annuity(
+                        capex=1, n=duration, wacc=m.discount_rate
```

Comment:
Shouldn't this `wacc` also be `interest` like in L1756

Reply:
Yeah, thank you. I might have overthought this:
* The annuity is calculated from an investors' point of view. You may assume the investor to be a "social planner" (whatever that shall be).
* Indeed, to be consistent, for converting it into a net present value, you want to stick to the investors' perspective (which I didn't). Mathematically, you could apply any interest rate here. I thought about only accounting for the overall model discount rate (I only use an assumed inflation rate here), but then I mix up the perspectives as you pointed out correctly.
* The (overall) discounting to t = 0 is already taken care about by the discount factor multiplier.

I'll fix this.